DOCS.md

```diff
@@ -0,0 +1,459 @@
+        dispatch(actions.user.reset());
+      }      
+    ]    
+  }
+}
+```
+
+####validation (data, callback)
+- @param **data** - response data
+  > type: Object
+
+- @param **callback** - you need to execute this callback function to finish data validation
+  > type: Function
+
+- @example
+```js
+{
+  test: {
+    url: "/api/test",
+    validation: (data, cb) {
+      // check data format
+      let error;
+      if (data instanceOf Array) {
+        error = "Data must be array";
+      }
+      cb(error);
+    }
+  }
+}
+```
+
+#### reducerName
+- @description:  Sometimes though, you might want named actions that go back to the same reducer. For example:
+- @type: String
+- @example:
+```js
+import reduxApi, {transformers} from "redux-api";
+const rest = reduxApi({
+  getUser: {
+    reducerName: "user"
+    url: "/user/1", // return a user object
+  }
+  updateUser: {
+    reducerName: "user"
+    url: "/user/1/update",
+    options: {
+      method: "post"
+    }
+  }
+});
+const {actions} = rest;
+
+// In component with redux support (see example section)
+const {dispatch} = this.props;
+dispatch(rest.actions.getUser()); // GET "/api/v1/entry"
+dispatch(rest.actions.updateUser({}, {
+  body: JSON.stringify({ name: "Hubot", login: "hubot"})
+}));  // POST "/api/v1/entry/1" with body
+
+```
+In the above example, both getUser, and updateUser update the same user reducer as they share the same reducerName
+For example used es7 javascript
+
+####helpers
+- @description: you can create custom helper function which work with this rest endpoint but with different parameters.
+- @type: Object
+- @example:
+```js
+{  
+  logger: "/api/logger",
+  test: {
+    url: "/api/test/:name/:id",
+    helpers: {
+      get(id, name) {
+        return [{id, name}], {}]
+      },
+      post(id, name, data) {
+        const {uuid} = this.getState().test;
+        const urlparams = {id, name};
+        const params = {body: {uuid, data}};
+        return [urlparams, params];
+      },
+      // complicated async logic
+      async() {
+        const {dispatch} = this;
+        return (cb)=> {
+          dispatch(rest.actions.logger((err)=> {
+            const args = [{id: 1, name: "admin"}];
+            cb(err, args);
+          }));
+        };
+      }
+    }
+  }
+}
+// using helpers
+rest.actions.test.get(1, "admin");
+// with callback
+rest.actions.test.post(1, "admin", {msg: "Hello"}, (err)=> {
+// end of action
+});
+rest.actions.test.async();
+```
+
+### reduxApi object
+
+####use(key, value)
+- @description initialize `reduxApi` with custom properties
+- @param **key** - name of property
+- @param **value** - value of property
+
+####list of properties
+####fetch 
+- @description backend adapter. In curent example we use `adaptersFetch` adapter for rest backend using `fetch` API for rest [isomorphic-fetch](https://www.npmjs.com/package/isomorphic-fetch)  
+- @example
+```js
+import adapterFetch from "redux-api/adapters/fetch";
```

Comment:
Great that this is documented! I need to use `redux-api/lib/adapters/fetch`, however, to get it to work.